### PR TITLE
Esil float

### PR DESCRIPTION
**Pull request by aemmitt-ns**
**Checklist**

- [x] Closing issues: #17541
- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

Add floating point operations. This is based on code by @FXTi in r2ghidra, slightly adapted to push the doubles to the stack as normal hex strings. This results in a lot less friction with the current working of ESIL

### CODE REVIEW

**Comment by aemmitt-ns** [comment]
Still working on this, I at least want to add support for `long double` if its available. I also can make the type conversions cleaner with some Unions. Ima do that now.

File: libr/anal/esil.c [hunk 1]
```diff
@@ -424,6 +430,18 @@ R_API bool r_anal_esil_pushnum(RAnalEsil *esil, ut64 num) {
 	return r_anal_esil_push (esil, str);
 }
 
+R_API bool r_anal_esil_pushbig(RAnalEsil *esil, ut8 *num, ut32 size) {
```
Comment (trufae) [inline libr/anal/esil.c:433]: looks unused and wrong

File: libr/anal/esil.c [hunk 2]
```diff
@@ -424,6 +430,18 @@ R_API bool r_anal_esil_pushnum(RAnalEsil *esil, ut64 num) {
...
+	RNumBig *big_num = r_big_new();
+	memcpy(big_num, num, size % R_BIG_ARRAY_SIZE);
+	return r_anal_esil_push (esil, r_big_to_hexstr(big_num));
+}
+
+// these numbers are jacked
+R_API RNumBig* r_anal_esil_get_big(RAnalEsil *esil, const char *str, RNumBig *big_num) {
+	r_big_from_hexstr(big_num, str);
```
Comment (trufae) [inline libr/anal/esil.c:441]: pleaes, follow the codingstyle (space before (, braces opening at right, ..), `} else {` in one line..

File: libr/anal/esil.c [hunk 3]
```diff
@@ -424,6 +430,18 @@ R_API bool r_anal_esil_pushnum(RAnalEsil *esil, ut64 num) {
...
+	return big_num;
+}
+
 R_API bool r_anal_esil_push(RAnalEsil *esil, const char *str) {
 	if (!str || !esil || !*str || esil->stackptr > (esil->stacksize - 1)) {
```
Comment (trufae) [inline libr/anal/esil.c:446]: use R_STR_ISEMPTY (str) and maybe use r_return_val_if_fail () for esil && R_STR_ISEMPTY (str), and do dynamic if for the stacksize comparision

File: libr/anal/esil.c [hunk 4]
```diff
@@ -2907,6 +2925,488 @@ static bool esil_set_delay_slot(RAnalEsil *esil) {
 	return ret;
 }
 
+static int esil_get_parm_float(RAnalEsil *esil, const char *str, double *num)
```
Comment (trufae) [inline libr/anal/esil.c:2928]: brace at right

**Review by trufae — CHANGES_REQUESTED** [review]
Can you add an example and some tests using `ae`?

File: libr/anal/esil.c (hunk 1 above)
Comment (aemmitt-ns) [reply libr/anal/esil.c:433]: This is going to be used for float80, float128 and larger multiplication results, I am working on it now.

Comment (aemmitt-ns) [reply libr/anal/esil.c:433]: I decided to leave this to another PR. So this PR will only support 32 and 64 bit floats.

File: libr/anal/esil.c (hunk 3 above)
Comment (aemmitt-ns) [reply libr/anal/esil.c]: this isn't from my PR but I can make these changes

Comment (trufae) [reply libr/anal/esil.c]: So its ready to merge?

**Pull request merged**